extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -378,37 +403,51 @@ public boolean apply(Throwable input)
               }
             }
 
-            final long nextOffset = record.offset() + 1;
-            final long endOffset = ioConfig.getEndPartitions().getPartitionOffsetMap().get(record.partition());
-
-            nextOffsets.put(record.partition(), nextOffset);
+            nextOffsets.put(record.partition(), record.offset() + 1);
+          }
 
-            if (nextOffset == endOffset && assignment.remove(record.partition())) {
-              log.info("Finished reading topic[%s], partition[%,d].", record.topic(), record.partition());
-              assignPartitions(consumer, topic, assignment);
-              stillReading = !assignment.isEmpty();
-            }
+          if (nextOffsets.get(record.partition()).compareTo(endOffsets.get(record.partition())) == 0
```

Comment:
Ah, at first I supported a case where the endOffsets could be set to less than the current offset and the compareTo was <= 0, and then I decided to instead enforce endOffsets >= currentOffset so I just changed it to == 0. Equals sounds good :)